src/legendhpges/icpc.py

```diff
@@ -0,0 +1,235 @@
+"""Build the inverted coaxial point contact crystal template."""
+import json
+import logging
+from math import pi, tan
+
+import numpy as np
+
+# Third-party imports
+import pyg4ometry as pg4
+
+
+class icpc:
+    """Define ICPC Germanium template."""
+
+    def __init__(self, jsonfile, reg, materials):
```

Comment:
I would like the constructor to accept both a `str` (JSON file name) or a `dict` (parsed JSON):
```py
def __init__(self, data: str | dict, ...) -> None:
  ...
```

Reply:
Why material*s* (plural)?

Reply:
I would also make `material` optional, defaulting to natural germanium.

Reply:
I haven't implemented the `str | dict` yet as Im not sure what the recommended way to handle different paths by type is with type annotations (I found `@singledispatch`, though not clear on its use in `__init__`).